### ISSUE TITLE
feat: pagination for notifications

### ISSUE THREAD

**Issue by vedant-jain03** [CircuitVerse/CircuitVerse#3329]
**Is your feature request related to a problem? Please describe.**
There may be more than 50 notifications for users which makes the webpage lengthy.

**Describe the solution you'd like**
Pagination for notification just like we have on the home page for "circuits". notifications per page = 10.

**Are you working on this? (Yes/No)**
No

**Comment by aashimawadhwa** [comment]
Hello @vedant-jain03  I am looking forward to work in this issue can you assign it to me and elaborate more about it?

**Comment by vedant-jain03** [comment]
> Hello @vedant-jain03 I am looking forward to work in this issue can you assign it to me and elaborate more about it?

Go ahead @aashimawadhwa

**Comment by Adityasaini10012001** [comment]
hey @aashimawadhwa are you working on this issue

**Comment by aashimawadhwa** [comment]
Hey, yes i am working on it, in the middle of it will finish by weekend .

**Comment by Mbiakop1** [comment]
Hi, @aashimawadhwa are you still working on this issue?

**Comment by vedant-jain03** [comment]
Hey @aashimawadhwa, any progress on this?

**Comment by aashimawadhwa** [comment]
> Hey @aashimawadhwa, any progress on this?

Hey I was sick and had exams in the middle I really want to work on it will finish it asap.

**Comment by pavanjoshi914** [comment]
nice catch @vedant-jain03 but do we really have that level of intensity in notifications that this feature will be in use frequently? 
if yes then it's good to go else it will be just a extra piece of code!

WDYT @tachyons ?

**Comment by vedant-jain03** [comment]
> nice catch @vedant-jain03 but do we really have that level of intensity in notifications that this feature will be in use frequently? if yes then it's good to go else it will be just a extra piece of code!
> 
> WDYT @tachyons ?

I would yes, as we have more notification events coming up!

**Comment by tachyons** [comment]
Yes, there will be more notifications.

**Comment by vedant-jain03** [comment]
@aashimawadhwa are you looking forward to work on this issue?

**Comment by vedant-jain03** [comment]
> @vedant-jain03 I would like to work on this issue. I've some ideas which will improve the User Experience. Can we have an Infinite scroll in the notification panel instead of pagination?

@Waishnav would you like to work on this issue?

**Comment by Waishnav** [comment]
Yeah sure.
But I've one doubt,
Do we need pagination ?? or do we need an infinite scroll in the notification section?

**Comment by Waishnav** [comment]
@vedant-jain03 We can implement it using flex_infinite_scroll gem easily.

**Comment by MitAbhay** [comment]
Hey @vedant-jain03, I would like to work on this issue, if it's not assigned to anyone.

**Comment by cAmartya** [comment]
If no one is working on this feature request. Can I be assigned to this?

**Comment by hussain-shariff** [comment]
@vedant-jain03 is this issue closed??

**Comment by vedant-jain03** [comment]
@cAmartya any updates?

**Comment by cAmartya** [comment]
> @cAmartya any updates?

I am interested to work on this issue, but I was not assigned the issue last time, so I did not start to work on this issue specifically.

**I want to work on this issue, Can you assign me this issue?**

**Comment by hussain-shariff** [comment]
@cAmartya hey I'm too looking forward to work on this issue, we can work together if you don't mind

**Comment by inviBeast** [comment]
@vedant-jain03 Hello I would be highly obliged if you assign me this issue i would love to contribute to circuit verse.

**Comment by hussain-shariff** [comment]
> @vedant-jain03 Hello I would be highly obliged if you assign me this issue i would love to contribute to circuit verse. 

@inviBeast hey can you look for some other issues, I have asked this to be assigned before you

**Comment by vedant-jain03** [comment]
@Waishnav Are you working on this?

**Comment by Waishnav** [comment]
No. I'm not. As off now I'm not working on any issue so, I would like to work on this one.

**Comment by vedant-jain03** [comment]
> No. I'm not. As off now I'm not working on any issue so, I would like to work on this one.

Go ahead!

**Comment by fillingtothemomo** [comment]
is this issue, still open i would like to contribute to it, can you assign it to me.
